pkg/core/kustomize/kustomize.go

```diff
@@ -0,0 +1,100 @@
+package kustomize
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/projectriff/riff/pkg/fileutils"
+	"io/ioutil"
+	"net/url"
+	"sigs.k8s.io/kustomize/k8sdeps"
+	"sigs.k8s.io/kustomize/pkg/commands/build"
+	"sigs.k8s.io/kustomize/pkg/fs"
+	"time"
+)
+
+type Kustomizer interface {
+	// Applies the provided label to the provided remote or local resource definition
+	// Returns the customized resource contents
+	// Returns an error if
+	// - the URL scheme is not supported (only file, http and https are)
+	// - retrieving the content fails
+	// - applying the customization fails
+	// As of the current implementation, it is not safe to call this function concurrently
+	ApplyLabel(resourceUri *url.URL, label *Label) ([]byte, error)
+}
+
+type Label struct {
```

Comment:
k8s just uses a `map[string]string`. This also gives us the ability to apply multiple labels.

Reply:
done